Failing to build on Windows

There are currently two issues with dependencies preventing a build on Windows.

1) The fuse library doesn't compile on Windows. Since it's used by the `mount-index` command only, and won't work on Windows anyway, disable the command similar to `tar`.
```
github.com/hanwen/go-fuse/fuse
/home/tesla/bin/pkg/mod/github.com/hanwen/go-fuse@v0.0.0-20180831115513-1d35017e9701/fuse/types.go:402:2: undefined: Attr
```

2) Determining the blocksize fails since the internal data structures are different. Blocksize is only used for reflinking which isn't supported on Windows so split up the file and return the default for all calls.
```
../../assemble.go:222:8: undefined: syscall.Stat_t
../../assemble.go:223:20: sys.Blksize undefined (type interface {} is interface with no methods)
```

Reported by @Jay-Detroja

--- a/assemble.go
+++ b/assemble.go
@@ -6,7 +6,6 @@ import (
 	"fmt"
 	"os"
 	"sync"
-	"syscall"
 
 	"github.com/pkg/errors"
 )
@@ -211,17 +210,4 @@ loop:
 
 	wg.Wait()
 	return stats, pErr
-}
-
-func blocksizeOfFile(name string) uint64 {
-	stat, err := os.Stat(name)
-	if err != nil {
-		return DefaultBlockSize
-	}
-	switch sys := stat.Sys().(type) {
-	case *syscall.Stat_t:
-		return uint64(sys.Blksize)
-	default:
-		return DefaultBlockSize
-	}
 }

--- a/blocksize.go
+++ b/blocksize.go
@@ -1,0 +1,21 @@
+// +build !windows
+
+package desync
+
+import (
+	"os"
+	"syscall"
+)
+
+func blocksizeOfFile(name string) uint64 {
+	stat, err := os.Stat(name)
+	if err != nil {
+		return DefaultBlockSize
+	}
+	switch sys := stat.Sys().(type) {
+	case *syscall.Stat_t:
+		return uint64(sys.Blksize)
+	default:
+		return DefaultBlockSize
+	}
+}

--- a/blocksize_windows.go
+++ b/blocksize_windows.go
@@ -1,0 +1,7 @@
+package desync
+
+func blocksizeOfFile(name string) uint64 {
+	// TODO: Not that it really matters for reflink cloning of files on windows
+	// but it would be nice to determine the actual blocksize here anyway.
+	return DefaultBlockSize
+}

--- a/cmd/desync/mount-index.go
+++ b/cmd/desync/mount-index.go
@@ -1,3 +1,5 @@
+// +build !windows
+
 package main
 
 import (

--- a/cmd/desync/mount-index_windows.go
+++ b/cmd/desync/mount-index_windows.go
@@ -1,0 +1,10 @@
+package main
+
+import (
+	"context"
+	"fmt"
+)
+
+func mountIdx(context.Context, []string) error {
+	return fmt.Errorf("Subcommand not available on this platform")
+}

--- a/mount-index.go
+++ b/mount-index.go
@@ -1,3 +1,5 @@
+// +build !windows
+
 package desync
 
 import (

